conjure-go-client/httpclient/response_error_decoder_middleware_test.go

```diff
@@ -65,7 +65,7 @@ func TestErrorDecoderMiddlewares(t *testing.T) {
 				assert.True(t, ok)
 				assert.Equal(t, 307, code)
 				location, ok := httpclient.LocationFromError(err)
-				assert.True(t, ok)
+				assert.False(t, ok)
```

Comment:
Since we are now using `http.Response.Location()` in the error decoder middleware, this test return false since there is no valid `location` header returned in this test.